gitindex/index.go

```diff
@@ -511,6 +520,29 @@ func IndexGitRepo(opts Options) error {
 	return builder.Finish()
 }
 
+func newIgnoreMatcher(tree *object.Tree) (*ignore.IgnoreMatcher, error) {
+	ignoreFile, err := tree.File(ignore.IgnoreFile)
+	if err == object.ErrFileNotFound {
+		return &ignore.IgnoreMatcher{}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	content, err := ignoreFile.Contents()
+	if err != nil {
+		return nil, err
+	}
+	pattern, err := ignore.ParseIgnoreFile(strings.NewReader(content))
```

Comment:
its a shame that we don't index anything if there is a typo here. We don't really have a good workflow here around telling the admin about it though. So both ignoring and failing have equal downsides. I suppose failing is more noticeable.